web-services-app/src/test/java/edu/unc/lib/boxc/web/services/processing/IiifV3ManifestServiceTest.java

```diff
@@ -0,0 +1,218 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+
+import java.util.Arrays;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.openMocks;
+
+/**
+ * @author bbpennel
+ */
+public class IiifV3ManifestServiceTest {
+    private static final String IIIF_BASE = "http://example.com/iiif/v3/";
+    private static final String SERVICES_BASE = "http://example.com/services/api/";
+    private static final String ACCESS_BASE = "http://example.com/";
+    private static final String WORK_ID = "5d72b84a-983c-4a45-8caa-dc9857987da2";
+    private static final String FILE1_ID = "faffb3e1-85fc-451f-9075-c60fc7584c7b";
+    private static final String FILE2_ID = "b6c51e59-d931-41d6-ba26-ec54ba9b2ef5";
+    private static final String COLL_ID = "fdce64cb-6a6f-43bb-8ed2-58f3b60148bf";
+    private static final PID WORK_PID = PIDs.get(WORK_ID);
+
+    @Mock
+    private AccessCopiesService accessCopiesService;
+    @Mock
+    private AccessControlService accessControlService;
+    @Mock
+    private AgentPrincipals agent;
+    @Mock
+    private AccessGroupSet principals;
+    private AutoCloseable closeable;
+
+    private ContentObjectSolrRecord workObj;
+
+    private IiifV3ManifestService manifestService;
+
+    @BeforeEach
+    public void setup() {
+        closeable = openMocks(this);
+        manifestService = new IiifV3ManifestService();
+        manifestService.setAccessCopiesService(accessCopiesService);
+        manifestService.setAccessControlService(accessControlService);
+        manifestService.setBaseIiifv3Path(IIIF_BASE);
+        manifestService.setBaseServicesApiPath(SERVICES_BASE);
+        manifestService.setBaseAccessPath(ACCESS_BASE);
+
+        when(agent.getPrincipals()).thenReturn(principals);
+
+        workObj = new ContentObjectSolrRecord();
+        workObj.setId(WORK_ID);
+        workObj.setResourceType(ResourceType.Work.name());
+        workObj.setTitle("Test Work");
+    }
+
+    private ContentObjectRecord createFileRecord(String id) {
+        var fileObj = new ContentObjectSolrRecord();
+        fileObj.setId(id);
+        fileObj.setResourceType(ResourceType.File.name());
+        fileObj.setTitle("File Object " + id);
+        var originalDs = new DatastreamImpl("original_file|image/jpeg|image.jpg|jpg|0|||240x750");
+        var jp2Ds = new DatastreamImpl("jp2|image/jp2|image.jp2|jp2|0|||");
+        fileObj.setDatastream(Arrays.asList(originalDs.toString(), jp2Ds.toString()));
+        return fileObj;
+    }
+
+    @Test
+    public void buildManifestNoViewableTest() {
```

Comment:
maybe change the test name to `buildManifestNoViewableFilesTest` to make it more clear